modules/consensus/diffs.go

```diff
@@ -288,15 +288,20 @@ func (cs *ConsensusSet) deleteObsoleteDelayedOutputMaps(bn *blockNode, dir modul
 func (cs *ConsensusSet) updateCurrentPath(bn *blockNode, dir modules.DiffDirection) {
 	// Update the current path.
 	if dir == modules.DiffApply {
-		cs.currentPath = append(cs.currentPath, bn.block.ID())
 		if cs.updatePath {
-			cs.db.pushPath(bn.block)
+			err := cs.db.pushPath(bn.block.ID())
+
+			if err != nil {
```

Comment:
One of the reasons for checking build.DEBUG is to remove the code when the flag isn't true. If you have it as part of a multipart conditional will this still be the case?